scheduler/pkg/store/pipeline/utils.go

```diff
@@ -200,6 +252,29 @@ func updateInputSteps(pipelineName string, inputs []string) []string {
 	return updatedInputs
 }
 
+func updateExternalInputSteps(inputs []string) []string {
```

Comment:
could we add more documentation and/or unit test to explain logic?

Reply:
will add test